geopandas/base.py

```diff
@@ -363,14 +363,98 @@ def count_coordinates(self):
         2    1
         3    5
         4    0
-        dtype: int64
+        dtype: int32
 
         See also
         --------
         GeoSeries.get_coordinates : extract coordinates as a :class:`~pandas.DataFrame`
+        GoSeries.count_geometries : count the number of geometries in a collection
         """
         return Series(self.geometry.values.count_coordinates(), index=self.index)
 
+    def count_geometries(self):
+        """
+        Returns a ``Series`` containing the count of geometries in a collection.
```

Comment:
```suggestion
        Returns a ``Series`` containing the count of geometries in each geometry.
```

Would that be confusing because it is using twice "geometry"? But on the other hand it is more consistent with the `count_coordinates` change you made and is clearer it is done _for each_ geometry/collection (and you explain below that non-collections will return 1)